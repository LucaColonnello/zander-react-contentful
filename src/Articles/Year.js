import React from 'react';
import ArticlesItemView from './ArticlesItemView';

const Year = props => {
  const { articles } = props;

  return (
    <div className="contentList-group">
      <aside className="contentList-group-year" role="complementary">
        {props.year}
      </aside>
      <ul className="contentList-group-items">
        {articles
          .filter(item => {
            return item.year === props.year;
          })
          .map(item => {
            const { title, subtitle, date } = item.fields;
            const id = item.sys.id;

            return (
              <ArticlesItemView
                title={title}
                subtitle={subtitle}
                date={date}
                id={id}
                key={id}
              />
            );
          })}
      </ul>
    </div>
  );
};

export default Year;
