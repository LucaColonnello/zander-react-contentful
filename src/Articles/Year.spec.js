import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Year from './Year';
import articleListing from '../state/articleListingStub';
const articles = articleListing;

test('Year view snapshot test', () => {
  const component = shallow(<Year year="2017" articles={articles} />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});
