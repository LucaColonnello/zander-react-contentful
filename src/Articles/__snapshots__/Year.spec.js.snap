// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Year view snapshot test 1`] = `
<div
  className="contentList-group"
>
  <aside
    className="contentList-group-year"
    role="complementary"
  >
    2017
  </aside>
  <ul
    className="contentList-group-items"
  />
</div>
`;
